baseplate/experiments/providers/single_variant.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+from .simple_experiment import SimpleExperiment
+
+
+class SingleVariantExperiment(SimpleExperiment):
+    """Basic experiment, handling more only a two variants (typically
+    one control, and one treatment).
+    This type of experiment allows the adjusting of variant sizes without
+    bucketing expisting users.
+
+    Does log bucketing events.
+
+    The config dict is expected to have the following values:
+
+        * **variants**: array of dicts, each containing the keys 'name'
+          and 'size'. Name is the variant name, and size is the fraction of
+          users to bucket into the corresponding variant. Sizes are expressed
+          as a floating point value between 0 and 1.
+        * **bucket_val**: (Optional) Name of the parameter you want to use for
+          bucketing.  This value must be passed to the call to
+          experiment.variant as a keyword argument.  Defaults to "user_id".
+        * **seed**: (Optional) Overrides the seed for this experiment.  If this
+          is not set, a combination of the experiment name, id, and shuffle
+          version is used.
+    """
+
+    @classmethod
+    def from_dict(cls, id, name, owner, start_ts, stop_ts, config,
+                  enabled=True, **kwargs):
+        return cls(
+            id=id,
+            name=name,
+            owner=owner,
+            start_ts=start_ts,
+            stop_ts=stop_ts,
+            enabled=enabled,
+            config=config,
+        )
+
+    def _validate_variants(self, variants):
+
+        super(SingleVariantExperiment, self)._validate_variants(variants)
+
+        if len(variants) != 2:
```

Comment:
probably want to verify that the sizes work out too, right?

Reply:
what do you mean by "sizes work out"?

Reply:
See comment below about super call, if you're referring to sum(sizes) < 1.0